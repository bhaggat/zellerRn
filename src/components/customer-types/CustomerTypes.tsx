import React from 'react';
import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
import RadioButton from '../radio-button/RadioButton';
import {CustomerType, customerTypeOptions} from '../../screens/home/Home';

type CustomerTypesProps = {
  selectedCustomerType: CustomerType;
  onSelectCustomerType: (customerType: CustomerType) => void;
};

const CustomerTypes = ({
  selectedCustomerType,
  onSelectCustomerType,
}: CustomerTypesProps) => {
  return (
    <View style={styles.optionsContainer}>
      {customerTypeOptions.map(option => {
        const isSelected = selectedCustomerType === option;
        return (
          <TouchableOpacity
            key={option}
            style={[
              styles.radioButtonContainer,
              isSelected ? styles.selectedButton : {},
            ]}
            onPress={() => onSelectCustomerType(option)}>
            <RadioButton isChecked={isSelected} />
            <Text style={styles.option}>{option}</Text>
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

const styles = StyleSheet.create({
  optionsContainer: {
    gap: 4,
  },
  radioButtonContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 12,
    borderRadius: 8,
    gap: 10,
  },
  selectedButton: {
    backgroundColor: '#E8F2FB',
  },
  option: {
    textTransform: 'capitalize',
    fontSize: 16,
  },
});

export default CustomerTypes;
