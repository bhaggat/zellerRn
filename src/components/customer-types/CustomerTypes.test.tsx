import {render} from '@testing-library/react-native';
import CustomerTypes from './CustomerTypes';

describe('Divider Component', () => {
  const onSelectMock = jest.fn();
  const role = 'ADMIN';
  describe('Snapshot', () => {
    test('matches', () => {
      const tree = render(
        <CustomerTypes
          selectedCustomerType={role}
          onSelectCustomerType={onSelectMock}
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Render', () => {
    test('renders correctly', () => {
      const tree = render(
        <CustomerTypes
          selectedCustomerType={role}
          onSelectCustomerType={onSelectMock}
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
