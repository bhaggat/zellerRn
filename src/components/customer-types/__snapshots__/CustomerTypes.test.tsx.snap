// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component Render Component render 1`] = `
<View
  style={
    {
      "gap": 4,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E8F2FB",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#E2E8F0",
            "borderWidth": 1,
            "justifyContent": "center",
          },
          {
            "borderRadius": 10,
            "height": 20,
            "width": 20,
          },
          {
            "borderColor": "#0370CE",
          },
          undefined,
        ]
      }
      testID="radioButton"
    >
      <View
        style={
          {
            "backgroundColor": "#0370CE",
            "borderRadius": 5,
            "height": 10,
            "width": 10,
          }
        }
        testID="innerRadioButton"
      />
    </View>
    <Text
      style={
        {
          "fontSize": 16,
          "textTransform": "capitalize",
        }
      }
    >
      ADMIN
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#E2E8F0",
            "borderWidth": 1,
            "justifyContent": "center",
          },
          {
            "borderRadius": 10,
            "height": 20,
            "width": 20,
          },
          false,
          undefined,
        ]
      }
      testID="radioButton"
    />
    <Text
      style={
        {
          "fontSize": 16,
          "textTransform": "capitalize",
        }
      }
    >
      MANAGER
    </Text>
  </View>
</View>
`;

exports[`Divider Component Render renders correctly 1`] = `
<View
  style={
    {
      "gap": 4,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E8F2FB",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#E2E8F0",
            "borderWidth": 1,
            "justifyContent": "center",
          },
          {
            "borderRadius": 10,
            "height": 20,
            "width": 20,
          },
          {
            "borderColor": "#0370CE",
          },
          undefined,
        ]
      }
      testID="radioButton"
    >
      <View
        style={
          {
            "backgroundColor": "#0370CE",
            "borderRadius": 5,
            "height": 10,
            "width": 10,
          }
        }
        testID="innerRadioButton"
      />
    </View>
    <Text
      style={
        {
          "fontSize": 16,
          "textTransform": "capitalize",
        }
      }
    >
      ADMIN
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#E2E8F0",
            "borderWidth": 1,
            "justifyContent": "center",
          },
          {
            "borderRadius": 10,
            "height": 20,
            "width": 20,
          },
          false,
          undefined,
        ]
      }
      testID="radioButton"
    />
    <Text
      style={
        {
          "fontSize": 16,
          "textTransform": "capitalize",
        }
      }
    >
      MANAGER
    </Text>
  </View>
</View>
`;

exports[`Divider Component Snapshot matches 1`] = `
<View
  style={
    {
      "gap": 4,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E8F2FB",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#E2E8F0",
            "borderWidth": 1,
            "justifyContent": "center",
          },
          {
            "borderRadius": 10,
            "height": 20,
            "width": 20,
          },
          {
            "borderColor": "#0370CE",
          },
          undefined,
        ]
      }
      testID="radioButton"
    >
      <View
        style={
          {
            "backgroundColor": "#0370CE",
            "borderRadius": 5,
            "height": 10,
            "width": 10,
          }
        }
        testID="innerRadioButton"
      />
    </View>
    <Text
      style={
        {
          "fontSize": 16,
          "textTransform": "capitalize",
        }
      }
    >
      ADMIN
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#E2E8F0",
            "borderWidth": 1,
            "justifyContent": "center",
          },
          {
            "borderRadius": 10,
            "height": 20,
            "width": 20,
          },
          false,
          undefined,
        ]
      }
      testID="radioButton"
    />
    <Text
      style={
        {
          "fontSize": 16,
          "textTransform": "capitalize",
        }
      }
    >
      MANAGER
    </Text>
  </View>
</View>
`;
